homeassistant/components/synology_dsm/switch.py

```diff
@@ -67,9 +75,9 @@ def __init__(
     @property
     def is_on(self) -> bool:
         """Return the state."""
-        return self.coordinator.data["switches"][self.entity_type]
+        return bool(self.coordinator.data["switches"][self.entity_type])
```

Comment:
With `coodinator` type specified, this `bool` shouldn't be needed.

Reply:
without `bool` mypy throws `switch.py:78: error: Returning Any from function declared to return "bool"  [no-any-return]` since  base class `SynologyDSMBaseEntity` defines `coordinator: DataUpdateCoordinator[dict[str, dict[str, Any]]]`

Reply:
Yes, you need to redefine type for self.coordinator as I described.

Reply:
I guess you mean this https://github.com/home-assistant/core/pull/49656#discussion_r624503193 ?

... while I'm writing this, I get how to add the definition :see_no_evil: